src/components/transport_manager/src/websocket_server/websocket_connection.cc

```diff
@@ -77,17 +75,15 @@ WebSocketConnection<WebSocketSecureSession<> >::WebSocketConnection(
     , session_(new WebSocketSecureSession<>(
           std::move(socket),
           ctx,
-          std::bind(
-              &WebSocketConnection::DataReceive, this, std::placeholders::_1),
-          std::bind(&WebSocketConnection::OnError, this)))
+          [this](Message frame) { DataReceive(frame); },
+          [this](Message frame) { DataSendDone(frame); },
+          [this](Message frame) { DataSendFailed(frame); },
+          [this]() { OnError(); }))
     , controller_(controller)
     , shutdown_(false)
     , thread_delegate_(new LoopThreadDelegate(
           &message_queue_,
-          std::bind(&WebSocketSecureSession<>::WriteDown,
-                    session_.get(),
-                    std::placeholders::_1),
-          std::bind(&WebSocketConnection::OnError, this)))
+          [this](Message frame) { return session_->WriteDown(frame); }))
```

Comment:
@mvorobio `return` is redundant in this case

Reply:
@AByzhynar please check https://github.com/LuxoftSDL/sdl_core/pull/18/commits/eb0fdf2e06a86f4111880459debd515831a5f2ec